File: Java Advanced/Java OOP/EXCEPTIONS AND ERROR HANDLING-Lab/src/validPerson/Person.java
package validPerson;

public class Person {
    private String  firstName;
    private String lastName;
    private int age;

    public Person(String firstName, String lastName, int age) {
        setFirstName(firstName);
        setLastName(lastName);
        setAge(age);
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        if(firstName.isBlank()){
            throw new IllegalArgumentException("The first name cannot be null");
        }
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        if(lastName.isEmpty() || lastName.equals("null")) {
            throw new IllegalArgumentException("The last name cannot be null");

        }
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if(age<0 || age>120) {
            throw new IllegalArgumentException("Age shoud be in age[0...120]");

        }
        this.age = age;
    }
}

File: Javascript/JS Applications/This-Lab/person.js
class Person{
    _first
    _last
    constructor(firstName,lastName){
        this._first=firstName;
        this._last=lastName;

    }

    get firstName(){
        return this._first
    }
    set firstName(name){
        this._first=name
    }


    get lastName(){
        return this._last
    }
    set lastName(name){
        this._last=name
    }

    get fullName(){
        return `${this.firstName} ${this.lastName}`
    }

    set fullName(name){
        let split=name.split(" ")
        if(split.length===2){
            this.firstName=split[0]
            this.lastName=split[1]
        }
    }
}
let person = new Person("Peter", "Ivanov");
console.log(person.fullName);//Peter Ivanov
person.firstName = "George";
console.log(person.fullName);//George Ivanov
person.lastName = "Peterson";
console.log(person.fullName);//George Peterson
person.fullName = "Nikola Tesla";
console.log(person.firstName)//Nikola
console.log(person.lastName)//Tesla
